lib/rules/one-var.js

```diff
@@ -62,20 +120,46 @@ module.exports = function(context) {
         "FunctionDeclaration": startFunction,
         "FunctionExpression": startFunction,
         "ArrowFunctionExpression": startFunction,
+        "BlockStatement": startBlock,
```

Comment:
You'll want `ForStatement` to cover `for` loops. The `switch` instance is interesting in that all the `case`s are within the same lexical environment, so `SwitchStatement` ought to work. (Technically the `discriminant` is outside the scope, but variable declarations can't occur immediately within that, so we're fine.)